source/server/connection_handler_impl.cc

```diff
@@ -184,6 +184,7 @@ void ConnectionHandlerImpl::ActiveSocket::unlink() {
 
 void ConnectionHandlerImpl::ActiveSocket::continueFilterChain(bool success) {
   if (success) {
+    bool no_error = true;
```

Comment:
Drive by: please add tests to connection_handler_impl_test specific to this new code. Thank you!

Reply:
Adding at my developing branch... Are you feeling this is the correct direction to address #7864 ?

Reply:
IMO this is a good solution for #7864 as I think it's simple and similar to what we do elsewhere. With that said I don't feel super strongly about it vs. adding a different filter return code. cc @PiotrSikora @lizan if you have opinions on this now is the time to weigh in. :)